client/client.go

```diff
@@ -304,6 +306,27 @@ func (c *Client) processChannelRequests() {
 			case *agentpb.StartActionRequest_PtSummaryParams:
 				action = actions.NewProcessAction(p.ActionId, c.cfg.Paths.PTSummary, []string{})
 
+			case *agentpb.StartActionRequest_PtMysqlSummaryParams:
+				// Parses the DSN string
+				cfg, err := mysql.ParseDSN(params.PtMysqlSummaryParams.Dsn)
+
+				// If not error
+				if err == nil {
+					// addr is IP:port
+					addr := cfg.Addr
+
+					// Strips the port info (if ':' found)
+					if i := strings.Index(addr, ":"); i > -1 {
+						addr = addr[:i]
+					}
+
+					// Action with path and --host, --user, --password data
+					action = actions.NewProcessAction(p.ActionId, c.cfg.Paths.PTMySqlSummary,
+						[]string{"--host=" + addr, "--user=" + cfg.User, "--password=" + cfg.Passwd})
```

Comment:
If MySQL is not running on the standard port, will this work sin port is missing here?

Reply:
Added the port to the arguments.